### DISCUSSION TITLE
RTL Support

### DISCUSSION THREAD

**Discussion by sagadsalem** [livewire/flux#196] [Feature Requests] [answered]
Hi Guys 

Flux is missing RTL support is there any date to expect to have RTL support on flux pro.

Thanks

**Comment by akn999** [comment]
I would like to add: why RTL support is important?

- It ensures that the components are fully accessible for right-to-left languages.
- Many developers working with multilingual websites need components that adapt to both LTR and RTL layouts without requiring significant customizations.
- RTL support would make this library even more versatile and usable in a global context.

**Comment by tao** [comment]
RTL support is a huge requirement and most issues should be quick to fix, here are some issues that appear from switch the docs to right to left direction.

---
![Screenshot 2024-10-12 at 10 28 02](https://github.com/user-attachments/assets/7d14ccbe-6b15-4608-98e5-d8844a969508)
---
![Screenshot 2024-10-12 at 10 19 10](https://github.com/user-attachments/assets/fa2ee92e-6f38-42fe-aef8-77a4ff69d1ae)
---
![Screenshot 2024-10-12 at 10 22 34](https://github.com/user-attachments/assets/60b03651-788f-42d8-bf7d-4c53711488b3)
---
![Screenshot 2024-10-12 at 10 32 37](https://github.com/user-attachments/assets/f8b85e56-40c6-4cdc-8cd2-160566fd7b37)
---
![Screenshot 2024-10-12 at 10 26 53](https://github.com/user-attachments/assets/3e3e0ad0-5560-49f2-ac20-0397e5019512)
---
![Screenshot 2024-10-12 at 10 29 31](https://github.com/user-attachments/assets/599f443e-a804-40af-8f56-f07f8c51f79c)
---
![Screenshot 2024-10-12 at 10 31 08](https://github.com/user-attachments/assets/6550eed7-5c4d-4b46-b8c4-f910837a9a05)
---
![Screenshot 2024-10-12 at 10 33 40](https://github.com/user-attachments/assets/83de834a-ff31-4298-87f3-dc614d3527a2)
---
![Screenshot 2024-10-12 at 10 35 07](https://github.com/user-attachments/assets/9ca58518-2741-40ea-b46f-7c0a15dfce5b)

**Reply by 3adeling** [reply]
> RTL support is a huge requirement and most issues should be quick to fix, here are some issues that appear from switch the docs to right to left direction.
> 
> ## ![Screenshot 2024-10-12 at 10 28 02](https://private-user-images.githubusercontent.com/1446331/378076299-7d14ccbe-6b15-4608-98e5-d8844a969508.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNzAyNzMsIm5iZiI6MTczNDM2OTk3MywicGF0aCI6Ii8xNDQ2MzMxLzM3ODA3NjI5OS03ZDE0Y2NiZS02YjE1LTQ2MDgtOThlNS1kODg0NGE5Njk1MDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTcyNjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTZjZDM2NWEzNThhODg0MzI0Y2E3MTM2OTlhYTY2Y2JlMDdjYjcyMWUwMTA3NzBkY2ZkZGM5YjcyMmZiOGMxYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.YlH1nEadmpJUndsb5KLD63Kl50hc9oLyKli-CSG6yEw)
> ## ![Screenshot 2024-10-12 at 10 19 10](https://private-user-images.githubusercontent.com/1446331/378076259-fa2ee92e-6f38-42fe-aef8-77a4ff69d1ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNzAyNzMsIm5iZiI6MTczNDM2OTk3MywicGF0aCI6Ii8xNDQ2MzMxLzM3ODA3NjI1OS1mYTJlZTkyZS02ZjM4LTQyZmUtYWVmOC03N2E0ZmY2OWQxYWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTcyNjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDc4NjdjN2RkYmQzZGUxYWVlMWQzOTgwMTk5ZDczNDliOGUxYzE1NWFhNGZhMjk5Yjg5NzUzNzViYTBiOWI2ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aMiSa_alroo0XED4pZ2T6BPrK9bYgqZNHostsKd9bZw)
> ## ![Screenshot 2024-10-12 at 10 22 34](https://private-user-images.githubusercontent.com/1446331/378076277-60b03651-788f-42d8-bf7d-4c53711488b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNzAyNzMsIm5iZiI6MTczNDM2OTk3MywicGF0aCI6Ii8xNDQ2MzMxLzM3ODA3NjI3Ny02MGIwMzY1MS03ODhmLTQyZDgtYmY3ZC00YzUzNzExNDg4YjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTcyNjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDdlNGYyZTJiMzU5MWE5NjAxZWNmOGJkZmFlOGZjNDk0YTNkYjBiZjVmNGRiNTVkYjg2MjA5Zjk5MTJlODY5NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Z6C5KwLZuMAEHf4G2q-aGgSXHoXjHo9FH3a1rkTc23g)
> ## ![Screenshot 2024-10-12 at 10 32 37](https://private-user-images.githubusercontent.com/1446331/378076339-f8b85e56-40c6-4cdc-8cd2-160566fd7b37.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNzAyNzMsIm5iZiI6MTczNDM2OTk3MywicGF0aCI6Ii8xNDQ2MzMxLzM3ODA3NjMzOS1mOGI4NWU1Ni00MGM2LTRjZGMtOGNkMi0xNjA1NjZmZDdiMzcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTcyNjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2FmNjEyOTdkMzRlNGQwOWVkZWYyMmUwZDc1Y2QwMGVlMGI5MWE5NTk3OTEzZmFjMzNjMzFiNzA4ODUwNGU5NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.jxjk6RukmqRaygFsbhMAJLqC8YTAUt0IwVsyLavHvqk)
> ## ![Screenshot 2024-10-12 at 10 26 53](https://private-user-images.githubusercontent.com/1446331/378076283-3e3e0ad0-5560-49f2-ac20-0397e5019512.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNzAyNzMsIm5iZiI6MTczNDM2OTk3MywicGF0aCI6Ii8xNDQ2MzMxLzM3ODA3NjI4My0zZTNlMGFkMC01NTYwLTQ5ZjItYWMyMC0wMzk3ZTUwMTk1MTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTcyNjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmM3OTNkMWQzZWFhMmFmNmQzYzlkNTdmOWNmNTIxYTZlZDMzYzM4NTdlNWE4M2JkZDc5ZDQyZTJhYzFhY2I0YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.wfSxmnmfxkR4q717WHR1EczjTJja-egc75wrYGhC84Q)
> ## ![Screenshot 2024-10-12 at 10 29 31](https://private-user-images.githubusercontent.com/1446331/378076307-599f443e-a804-40af-8f56-f07f8c51f79c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNzAyNzMsIm5iZiI6MTczNDM2OTk3MywicGF0aCI6Ii8xNDQ2MzMxLzM3ODA3NjMwNy01OTlmNDQzZS1hODA0LTQwYWYtOGY1Ni1mMDdmOGM1MWY3OWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTcyNjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTJlNjZhMzViZDAzZTA5Y2FhNzljZTdmZDEzY2I2NjE3MWVkMDNlZjZlODJmZTFjODJkNGU3ODUyYTI0NGRmZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Ia2Fnm0rRN6gHzAiH5FoRS2iqyMLZhW86W84OS_w1EA)
> ## ![Screenshot 2024-10-12 at 10 31 08](https://private-user-images.githubusercontent.com/1446331/378076329-6550eed7-5c4d-4b46-b8c4-f910837a9a05.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNzAyNzMsIm5iZiI6MTczNDM2OTk3MywicGF0aCI6Ii8xNDQ2MzMxLzM3ODA3NjMyOS02NTUwZWVkNy01YzRkLTRiNDYtYjhjNC1mOTEwODM3YTlhMDUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTcyNjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTkxMzVmZWU4YzliM2NiOWFlMmI2NDhmYzQzZDViZjkzZGY0ZTI4OGNiNzM2MmUzZTE1MWQwYTZjMTcxMzFhNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.qBpyuII55TY7gqt8rma5Bml38TF6dUCtzweC2LE4WBU)
> ## ![Screenshot 2024-10-12 at 10 33 40](https://private-user-images.githubusercontent.com/1446331/378076345-83de834a-ff31-4298-87f3-dc614d3527a2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNzAyNzMsIm5iZiI6MTczNDM2OTk3MywicGF0aCI6Ii8xNDQ2MzMxLzM3ODA3NjM0NS04M2RlODM0YS1mZjMxLTQyOTgtODdmMy1kYzYxNGQzNTI3YTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTcyNjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2U4ZjQ4Yjg3ZTRhYWU0MzlhZmNiOTIwOTFlOGE2MDVkMDljYmRjNGY1ZjNjYTM5NzFiYTFiN2ZiYjQ3NGE3OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9MfZ-TkcMpr1GH_C7gFe4X0BNPO6UdW9aa8kx4qx7-U)
> ![Screenshot 2024-10-12 at 10 35 07](https://private-user-images.githubusercontent.com/1446331/378076348-9ca58518-2741-40ea-b46f-7c0a15dfce5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNzAyNzMsIm5iZiI6MTczNDM2OTk3MywicGF0aCI6Ii8xNDQ2MzMxLzM3ODA3NjM0OC05Y2E1ODUxOC0yNzQxLTQwZWEtYjQ2Zi03YzBhMTVkZmNlNWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTcyNjEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTY5NDVhMDRjMDBlZmIyNTllZjJiMDJjNWIxYTFmNDUyN2ZiMzA5ODNjMjIzMDQ0ZjA1MjVkZmFhYjQ5N2I5NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.alC67zud6aXnesH9jirR6uNIwldw1n-4-vzgCkqXJ84)

RTL is not hard for flux. it is easy and I did everything in like 30 mins max. but the problem is that you to modify published components to apply it. the main problem is that @calebporzio is not listening to us!

**Reply by joshhanley** [reply]
@3adeling please don't think we aren't listening, we definitely are. Caleb and I have spoken about RTL support, but we can't prioritise everything all at once.

**Reply by 3adeling** [reply]
I appreciate your efforts, @joshhanley and @calebporzio. However, this feature is essential for us, and currently, Flux is unusable because it doesn't utilize the appropriate Tailwind classes that make RTL implementation effortless. RTL isn't complicated or rocket science; it's simply a small adjustment in how CSS classes are applied.

If RTL support isn't considered from the early stages, it can become difficult to implement later. That said, it doesn't need to be a concern if the correct classes are used from the start. I can provide you with all the necessary steps to ensure Flux is RTL-ready, and these practices should be followed consistently moving forward.

**Reply by joshhanley** [reply]
@3adeling sorry it's not suitable for you. I can't say if or when this is going to be implemented, so if you'd like a refund please email support@fluxui.dev and we can get that sorted for you.

**Reply by 3adeling** [reply]
This wasn’t the response I was expecting. I wasn’t planning to request a refund initially, but since you brought it up, I’ll go ahead and do so now. Thank you.

**Comment by 3adeling** [comment]
I don’t expect RTL support to be available anytime soon, so I won’t be using Flux in production for that reason.

**Comment by 3adeling** [comment]
We can assist with RTL support if needed @calebporzio @joshhanley.

**Comment by aliqasemzadeh** [comment]
Just we have to use start instead of right for paddings and margins. (end for left) Also no text alignments.
Laravel can add language direction in translates variables.

**Reply by 3adeling** [reply]
> Just we have to use start instead of right for paddings and margins. (end for left) Also no text alignments. Laravel can add language direction in translates variables.

<html><head></head><body><p>It's not just about the start and end; it's about considering RTL from the very beginning.</p>
<ol>
<li>
<p>Use <code inline="">ms-*</code>, <code inline="">me-*</code>, <code inline="">ps-*</code>, <code inline="">pe-*</code> instead of <code inline="">ml-*</code>, <code inline="">mr-*</code>, <code inline="">pl-*</code>, <code inline="">pr-*</code>.</p>
</li>
<li>
<p>Use <code inline="">text-start</code>, <code inline="">text-end</code> instead of <code inline="">text-left</code>, <code inline="">text-right</code>.</p>
</li>
<li>
<p>Use <code inline="">start-*</code>, <code inline="">end-*</code> instead of <code inline="">left-*</code>, <code inline="">right-*</code>.</p>
</li>
<li>
<p><strong>For border-radius adjustments:</strong></p>

Class | Left-to-Right | Right-to-Left
-- | -- | --
rounded-s-* | rounded-l-* | rounded-r-*
rounded-e-* | rounded-r-* | rounded-l-*
rounded-ss-* | rounded-tl-* | rounded-tr-*
rounded-se-* | rounded-tr-* | rounded-tl-*
rounded-es-* | rounded-bl-* | rounded-br-*
rounded-ee-* | rounded-br-* | rounded-bl-*


</li>
<li>
<p>Use <code inline="">rtl:space-x-reverse</code> wherever <code inline="">space-x-*</code> is applied.</p>
</li>
<li>
<p>Instead of <code inline="">translate-x-*</code> or <code inline="">-translate-x-*</code>, use <code inline="">ltr:translate-x-* rtl:-translate-x-*</code> or <code inline="">ltr:-translate-x-* rtl:translate-x-*</code>. This requires setting the <code inline="">dir</code> attribute on the HTML root element (see next point).</p>
</li>
<li>
<p>These changes require setting <code inline="">dir="ltr"</code> by default. To enable RTL styling dynamically, use:</p>
<pre><code class="language-html">dir="{{ app()-&gt;locale() === 'ar' ? 'rtl' : 'ltr' }}"
</code></pre>
</li>
</ol></body></html>

**Reply by 3adeling** [reply]
also in flux and flex-pro files alot of changes has to be done since right and left is being used there intensively

**Comment by joshhanley** [comment, marked as answer]
RTL support has been added in Flux v2.1.0
